File: src/Java/myservicebus/src/main/java/com/myservicebus/TransportCapabilitySupport.java
package com.myservicebus;

import com.fasterxml.jackson.annotation.JsonValue;

public enum TransportCapabilitySupport {
    NATIVE("native"),
    EMULATED("emulated"),
    UNSUPPORTED("unsupported");

    private final String protocolValue;

    TransportCapabilitySupport(String protocolValue) {
        this.protocolValue = protocolValue;
    }

    @JsonValue
    public String protocolValue() {
        return protocolValue;
    }
}

File: src/MyServiceBus/TransportCapabilitySupport.cs
using System.Text.Json.Serialization;

namespace MyServiceBus;

[JsonConverter(typeof(JsonStringEnumConverter<TransportCapabilitySupport>))]
public enum TransportCapabilitySupport
{
    [JsonStringEnumMemberName("native")]
    Native,
    [JsonStringEnumMemberName("emulated")]
    Emulated,
    [JsonStringEnumMemberName("unsupported")]
    Unsupported
}
